stdlib/public/Concurrency/PartialAsyncTask.swift

```diff
@@ -35,6 +52,72 @@ public struct UnownedJob: Sendable {
   }
 }
 
+@available(SwiftStdlib 5.9, *)
+extension UnownedJob {
+  public struct Priority {
+    public typealias RawValue = UInt8
+    public var rawValue: RawValue
+
+    /// Convert this ``UnownedJob/Priority`` to a ``TaskPriority``.
+    public var asTaskPriority: TaskPriority? {
+      // TODO: what if it was JobPriority::Undefined?
+      TaskPriority(rawValue: self.rawValue)
+    }
+  }
+}
+
+/// A unit of scheduleable work.
+///
+/// Unless you're implementing a scheduler,
+/// you don't generally interact with jobs directly.
+@available(SwiftStdlib 5.9, *)
+@frozen
+@_moveOnly
+public struct Job: Sendable {
+  internal var context: Builtin.Job
+
+  @usableFromInline
+  internal init(context: __owned Builtin.Job) {
+    self.context = context
+  }
+
+  public var priority: UnownedJob.Priority {
+    // let raw = _swift_concurrency_jobPriority(UnownedJob(context: self.context)) // TODO: do we also surface this or the base priority?
+    let raw = _taskCurrentPriority(context as! Builtin.NativeObject)
+    return UnownedJob.Priority(rawValue: raw)
+  }
+
+  // TODO: move only types cannot conform to protocols, so we can't conform to CustomStringConvertible;
+  //       we can still offer a description to be called explicitly though.
+  public var description: String {
+    "\(Self.self)(\(_getJobTaskId(UnownedJob(context: self.context)))"
```

Comment:
Really want to give people a chance to understand their scheduling and loop it back to what they'd see in instruments and swift-inspect -- thus printing with the task identifier. Yay, nay @Harjas12 @mikeash ?

Reply:
Printing the task ID seems good. If I'm remembering and reading the code correctly, non-Task Jobs will have an ID of 0, and Tasks always have a non-zero ID (it's actually a bit more strict than that, the bottom 32 bits will always contain a non-zero value) so they won't conflict. It might be a bit strange to always show 0 for non-Task Jobs though. Maybe have this thing return an Optional instead, with `nil` for plain Jobs, and don't print the ID in that case.

Reply:
That sounds good to me, printing nil when 0 I guess is more "understandable" for someone looking for "which job is doing something" -- if they're all `0` they might be led to believe it is the same magical "zero job" somehow. Will do 👍